app/controllers/events/view/sessions/create.js

```diff
@@ -1,13 +1,20 @@
 import Controller from '@ember/controller';
 export default Controller.extend({
   actions: {
-    async save() {
+    async save(speakerDetails) {
       try {
         this.set('isLoading', true);
-        await this.get('model.speaker').save();
-        this.get('model.speaker.sessions').pushObject(this.get('model.session'));
+        if (!speakerDetails) {
+          await this.get('model.speaker').save();
+        }
+        await this.get('model.session').save();
+        if (!speakerDetails) {
+          this.get('model.speaker.sessions').pushObject(this.get('model.session'));
+          await this.get('model.session').save();
```

Comment:
Why are we saving model.session twice or possibly three times?

Reply:
@Kreijstal initially in the session model the information about the session entered by the user in the form is saved. Then as the session has a relationship with speakers the model of speakers is added in the relationship(the existing one or the new one) and then the model is again saved.